src/main/java/MealDemo/rest/MealPlanningRESTController.java

```diff
@@ -72,5 +76,43 @@ public void updateMeal(@RequestBody Meal theMeal, @PathVariable int mealId){
         }
     }
 
+
+    @GetMapping("/meals/categories")
+    public Map<String, List<Meal>> getMealsByCategories(){
+        return mealPlanningServiceImpl.mealsByCategories();
+    }
+
+    @GetMapping("/meals/weekly-menu")
+    public HashMap<DaysOfTheWeek, Meal> generateWeeklyMeals(@RequestBody HashMap<String, Integer> userInput){
+
+        if (!userInput.isEmpty()){
+            System.out.println("Using custom values");
+            int sumRequest = 0;
+            for (int requestNumber : userInput.values()){
+                sumRequest += requestNumber;
+            }
+
+            if (sumRequest != 7){
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid Request: incorrect number of days requested");
```

Comment:
Technically the message should be:

> Invalid Request: incorrect number of meals requested. You should request 7 meals. One for each day of the week.

I.e. the user is requesting meals not days.